### DISCUSSION TITLE
./build_pag.sh  执行失败

### DISCUSSION THREAD

**Discussion by guileryu** [Tencent/libpag#2104] [Q&A] [answered]
linux 执行./build_pag.sh失败
![image](https://github.com/Tencent/libpag/assets/59436015/0a51f611-ee9e-49da-86cc-cbd0558094af)

node版本如下
![image](https://github.com/Tencent/libpag/assets/59436015/4910a246-b7ce-48eb-bf67-d751745155b2)

**Comment by guileryu** [comment]
@domchen

**Reply by domchen** [reply, marked as answer]
这是你那台电脑访问不了github的问题，可以在项目根目录手动执行一下depsync命令，第三方依赖都下载下来了没有报错再编译。也可以在本地电脑执行完depsync同步后整体打包到服务器端再编译

**Reply by guileryu** [reply]
![image](https://github.com/Tencent/libpag/assets/59436015/1adf06b2-43db-4a28-99ef-e69d69cbf1fc)

cmake --build ./build -- -j 12 最后一步执行失败是啥问题 @domchen

**Reply by domchen** [reply]
你前一步 build_pag.sh没有成功，成功一步再执行下一步

**Comment by longjie2014** [comment]
![image](https://github.com/Tencent/libpag/assets/7840917/7f4b927a-80fa-4c43-99b6-f2fcf3229f5b)
![image](https://github.com/Tencent/libpag/assets/7840917/bd0e6fd1-122d-4305-ad79-11273fb3fc7e)
我在根目录执行了depsync，所有三方依赖都确认下载完成，但是仍然报错，不知具体原因。

**Reply by domchen** [reply]
提示不是很明确吗？你要把整个libpag仓库传上去，包括里面的.git子目录，你现在这个libpag不是一个有效的git仓库

**Reply by longjie2014** [reply]
我已经把整个github上的所有代码上传，仍然报错，我的服务器操作系统为:centos 7.9,r所有的依赖都已经下载完成。
![image](https://github.com/Tencent/libpag/assets/7840917/27444d18-aaf6-416e-9700-7401e8080fa0)

**Reply by domchen** [reply]
你 ls 出来的结果不就是缺少 `.git` 目录吗？你没有上传隐藏的 libpag/.git 文件夹。可能上传的时候被你的工具软件过滤了。你可以打个zip包上传到服务器解压。如果服务器能联网直接在服务器上检出libpag仓库，然后depsync同步。

**Reply by domchen** [reply]
你现在卡住的都是非常基础的 git 使用问题，还没涉及到 PAG 编译的内容。如果根据报错提示都没法自己解决，后面只会遇到更多难解决的问题。建议还是请教一下有 C++ 编译经验的同事现场帮助比较快。
